docs/README.md

```diff
@@ -0,0 +1,36 @@
+# Kubernetes vSphere Cloud Provider
+
+This is the official documentation for the Kubernetes vSphere Cloud Provider.
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Concepts](#concepts)
+  - [Cloud Providers](#cloud-providers)
+  - [In-Tree Cloud Providers](#in-tree-cloud-providers)
+  - [Out-of-Tree Cloud Providers](#out-of-tree-cloud-providers)
+- [Components and Tools](#components-and-tools)
+  - [VM](#VM)
+  - [vSphere](#vsphere)
+  - [ESXi](#esxi)
+  - [vCenter](#vcenter)
+  - [govmomi](#govmomi)
+  - [Kubernetes](#kubernetes)
+  - [kube-apiserver](#kube-apiserver)
+  - [kubelet](#kubelet)
+  - [kube-controller-manager](#kube-controller-manager)
+  - [cloud-controller-manager](#cloud-controller-manager)
+  - [kubeadm](#kubeadm)
+- [Architecture](#architecture)
+  - [Kubernetes using the In-Tree vSphere Provider](#kubernetes-using-the-in-tree-vsphere-provider)
+  - [Kubernetes using the Out-of-Tree vSphere Provider](#kubernetes-using-the-out-of-tree-vsphere-provider)
+- [vSphere Integrations](#vsphere-integrations)
+  - [Kubernetes Nodes](#kubernetes-nodes)
+  - [Kubernetes Zones/Regions Topology](#kubernetes-zones-regions-topology)
+  - [Kubernetes LoadBalancers](#kubernetes-loadbalancers)
+  - [Kubnernetes Routes](#kubernetes-routes)
+- [Example Configurations and Manifests](#example-manifests)
+- [Tutorials](#tutorials)
+- [Addons](#addons)
+  - [Storage](#storage)
```

Comment:
I like the callouts for the addons. Let's just make sure that when we do fill this with content, we make it clear that the out-of-tree provider *requires* the CSI plugin.

Reply:
Good point, how about renaming this to `Required Plugins` and linking out to CSI docs?

Reply:
Sounds good -- we may want both categories: optional and required?

Reply:
How about `Required` and `Recommended`? I feel too many things fall into optional plugins 😆